workflow/controller/operator.go

```diff
@@ -1362,14 +1362,82 @@ func getTemplateOutputsFromScope(tmpl *wfv1.Template, scope *wfScope) (*wfv1.Out
 	return &outputs, nil
 }
 
+func HasOutputResultRef(name string, parentTmpl *wfv1.Template) bool {
+
+	nodeName := "." + name + "."
+
+	if parentTmpl.DAG != nil {
+		for _, dagTask := range parentTmpl.DAG.Tasks {
+
+			if strings.Contains(dagTask.When, nodeName) {
+				return true
+			}
+
+			for _, artf := range dagTask.Arguments.Parameters {
+				if strings.Contains(*artf.Value, "{{") && (strings.Contains(*artf.Value, nodeName) || (artf.ValueFrom != nil && strings.Contains(artf.ValueFrom.Parameter, nodeName))) {
+					return true
+				}
+			}
+		}
+	}
+
+	if parentTmpl.Steps != nil {
+		for idx := range parentTmpl.Steps {
+			stepGroup := parentTmpl.Steps[idx]
+			for _, step := range stepGroup {
+
+				if strings.Contains(step.When, nodeName) {
+					return true
+				}
+				for _, artf := range step.Arguments.Parameters {
+					if strings.Contains(*artf.Value, "{{") && (strings.Contains(*artf.Value, nodeName) || (artf.ValueFrom != nil && strings.Contains(artf.ValueFrom.Parameter, nodeName))) {
+						return true
+					}
+				}
+			}
+		}
+	}
+
+	for _, param := range parentTmpl.Outputs.Parameters {
+		if strings.Contains(*param.Value, "{{") && (strings.Contains(*param.Value, nodeName) || (param.ValueFrom != nil && strings.Contains(param.ValueFrom.Parameter, nodeName))) {
+			return true
+		}
+	}
+
+	return false
+}
+
+func GetStepOrDAGTaskName(nodeName string) string {
+	if strings.Contains(nodeName, ".") {
+		name := nodeName[strings.LastIndex(nodeName, ".")+1:]
+		// Check retry scenario
+		if strings.Contains(name, "(") {
```

Comment:
We don't need to do the parenthesis hack. We already have the template information and already know if it will be retried ahead of time.

Reply:
I think we need to handle the retry scenario with "(". This function is getting `NodeName: scripts-bash-g7vnp[0].generate(1)`.  We need to get step/task name from nodename.

Reply:
resolved it